src/Microsoft.Framework.PackageManager/List/AssemblyDependencyWalker.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Framework.PackageManager.Algorithm;
+using Microsoft.Framework.Runtime;
+using NuGet;
+
+namespace Microsoft.Framework.PackageManager.List
+{
+    public class AssemblyDependencyWalker
+    {
+        private readonly static FrameworkName AspNetCore50 = VersionUtility.ParseFrameworkName("aspnetcore50");
+
+        private readonly DFPreOrderWalker<string> _walker;
+        private readonly GacDependencyResolver _gacResolver;
+        private readonly FrameworkName _framework;
+        private readonly ApplicationHostContext _hostContext;
+        private readonly string _runtimeFolder;
+
+        public AssemblyDependencyWalker(
+            FrameworkName framework,
+            ApplicationHostContext hostContext,
+            string runtimeFolder)
+        {
+            _framework = framework;
+            _hostContext = hostContext;
+            _runtimeFolder = runtimeFolder;
+
+            Assemblies = new HashSet<string>();
```

Comment:
Might want to specify a Comparer. Either `StringComparer.Ordinal` \ `OrdinalIgnoreCase`